lxd/instance/drivers/driver_qemu_templates.go

```diff
@@ -93,193 +93,239 @@ func qemuBaseSections(architecture string) []cfgSection {
 		})
 }
 
-var qemuMemory = template.Must(template.New("qemuMemory").Parse(`
-# Memory
-[memory]
-size = "{{.memSizeBytes}}M"
-`))
+func qemuMemorySections(memSizeMB int64) []cfgSection {
+	return []cfgSection{{
+		name:    "device \"memory\"",
+		comment: "Memory",
+		entries: []cfgEntry{{key: "size", value: fmt.Sprintf("%dM", memSizeMB)}},
+	}}
+}
 
-var qemuSerial = template.Must(template.New("qemuSerial").Parse(`
-# Virtual serial bus
-[device "dev-qemu_serial"]
-{{- if eq .bus "pci" "pcie"}}
-driver = "virtio-serial-pci"
-bus = "{{.devBus}}"
-addr = "{{.devAddr}}"
-{{- end}}
-{{if eq .bus "ccw" -}}
-driver = "virtio-serial-ccw"
-{{- end}}
-{{if .multifunction -}}
-multifunction = "on"
-{{- end }}
+func qemuDeviceCommonEntries(
```

Comment:
Please define function signatures on single line.
If it has too many arguments that its getting unwieldy then this is s sign the function is taking too many arguments and may benefit  from taking a dedicated args struct type so that the fields being passed in are clearer to read.

Reply:
What would you say is a good threshold to move function arguments into a struct? I can see that some of these `qemu*Sections` will have a high number of arguments, should I switch all of them to receive structs instead of passing individual args?

Reply:
I don't think we have an absolute cut off on this.

Personally I prefer passing an arg struct in for scenarios when lots of args are in use, as it makes it more readable. Otherwise you end up with a a call like "foo(true, false, 3, " bar", nil)" which can be hard to understand.

Using a struct arg can make thins more readable e.g
 https://github.com/lxc/lxd/blob/master/lxd/network/driver_ovn.go#L2057

This was one advantage to the previous template approach so I certainly wouldn't be against using a struct type for each function.